cfme/tests/configure/test_access_control.py

```diff
@@ -2101,11 +2120,43 @@ def test_tenant_visibility_miq_ae_namespaces_all_parents():
     pass
 
 
-@pytest.mark.manual
+@pytest.fixture()
+def create_user(appliance):
+    """This fixture will create user, group, and role with all product feature
+    by clicking individually"""
+    def _create_user():
+        features = FEATURES_511 if appliance.version > "5.11" else FEATURES_510
+        product_features = [[["Everything", feature], True] for feature in features]
+        role = appliance.collections.roles.create(
+            name=fauxfactory.gen_alpha(15, start="API-role-"),
+            product_features=[[["Everything"], False]] + product_features,
+        )
+        # creating group
+        group = appliance.collections.groups.create(
+            description=fauxfactory.gen_alphanumeric(22, "group_description_"),
+            role=role.name,
+        )
+
+        creds = Credential(
+            principal=fauxfactory.gen_alphanumeric(start="user_"),
+            secret=fauxfactory.gen_alphanumeric(),
+        )
+        # user with above group
+        user = appliance.collections.users.create(
+            name=fauxfactory.gen_alphanumeric(start="user_"),
+            credential=creds,
+            email=fauxfactory.gen_email(),
+            groups=group,
+        )
+        return role, group, user
+    return _create_user
```

Comment:
Why is this fixture returning a method, delaying the actual creation of the role, group, and user to the test call phase?

There are some cases where this structure is relevant, but I don't think this is one of them.

The test steps include the creation of the custom role, but I think that could be considered part of test setup, where the actual test content is attempting to edit tags with that user+group+role combo.

I see John suggested above that it should be a fixture, and I'd agree with him - but if its a fixture, it should be yielding the role, group, and user to the test, and then cleaning up after the yield.

Reply:
Done, Moved to fixture